tsdb/engine/tsm1/iterator.gen.go.tmpl

```diff
@@ -108,7 +108,7 @@ type {{.name}}Iterator struct {
 	aux   []cursorAt
 	conds struct {
 		names []string
-		curs  []*bufCursor
+		curs  []cursorAt
```

Comment:
My biggest concern with this is that it requires an interface conversion at the lowest level of the query engine. It also precludes us from changing `curs` to a `[]bufCursor` to reduce allocations. I'm ok with this change now but we might need to revisit it in the future if it's slowing down queries significantly.